include/AudioBufferView.h

```diff
@@ -0,0 +1,256 @@
+	{
+	}
+
+	//! Construct dynamic channel count from static
+	template<proc_ch_t otherChannels>
+		requires (channelCount == DynamicChannelCount && otherChannels != DynamicChannelCount)
+	constexpr InterleavedBufferView(InterleavedBufferView<SampleT, otherChannels> other) noexcept
+		: Base{other.data(), otherChannels, other.frames()}
+	{
+	}
+
+	constexpr auto empty() const noexcept -> bool
+	{
+		return !this->m_data || this->channels() == 0 || this->m_frames == 0;
+	}
+
+	//! Returns the frame at the given index
+	constexpr auto frame(f_cnt_t index) const noexcept
+	{
+		if constexpr (channelCount == DynamicChannelCount)
+		{
+			return std::span<SampleT>{framePtr(index), this->channels()};
+		}
+		else
+		{
+			return std::span<SampleT, channelCount>{framePtr(index), this->channels()};
+		}
+	}
+
+	//! Returns the frame at the given index
+	constexpr auto framePtr(f_cnt_t index) const noexcept -> SampleT*
+	{
+		assert(index < this->m_frames);
+		return this->m_data + index * this->channels();
+	}
+};
+
+// Check that the std::span-like space optimization works
+static_assert(sizeof(InterleavedBufferView<float>) > sizeof(InterleavedBufferView<float, 2>));
+static_assert(sizeof(InterleavedBufferView<float, 2>) == sizeof(void*) + sizeof(f_cnt_t));
+
+
+/**
+ * Non-owning view for multi-channel non-interleaved audio data
+ *
+ * The data type is `SampleT* const*` which is a 2D array accessed as data[channel][frame index]
+ * where each channel's buffer contains `frames()` frames.
+ *
+ * TODO C++23: Use std::mdspan?
+ */
+template<typename SampleT, proc_ch_t channelCount = DynamicChannelCount>
+class PlanarBufferView : public detail::BufferViewData<SampleT* const, channelCount>
+{
+	using Base = detail::BufferViewData<SampleT* const, channelCount>;
+
+public:
+	using Base::Base;
+
+	//! Contruct const from mutable (static channel count)
+	template<typename T = SampleT> requires (std::is_const_v<T> && channelCount != DynamicChannelCount)
+	constexpr PlanarBufferView(PlanarBufferView<std::remove_const_t<T>, channelCount> other) noexcept
+		: Base{other.data(), other.frames()}
+	{
+	}
+
+	//! Contruct const from mutable (dynamic channel count)
+	template<typename T = SampleT> requires (std::is_const_v<T> && channelCount == DynamicChannelCount)
+	constexpr PlanarBufferView(PlanarBufferView<std::remove_const_t<T>, channelCount> other) noexcept
+		: Base{other.data(), other.channels(), other.frames()}
+	{
+	}
+
+	//! Construct dynamic channel count from static
+	template<proc_ch_t otherChannels>
+		requires (channelCount == DynamicChannelCount && otherChannels != DynamicChannelCount)
+	constexpr PlanarBufferView(PlanarBufferView<SampleT, otherChannels> other) noexcept
+		: Base{other.data(), otherChannels, other.frames()}
+	{
+	}
+
+	constexpr auto empty() const noexcept -> bool
+	{
+		return !this->m_data || this->channels() == 0 || this->m_frames == 0;
+	}
+
+	//! Returns the buffer of a given channel
```

Comment:
I have started to prefer using doxygen stuff like `@return` instead. I think it might make a difference when we finally get around to generating documentation?

Reply:
Done